docs/style.md

```diff
@@ -539,6 +539,245 @@ import { unstable_Pagination as Pagination } from 'carbon-components-react';
 This code should be treated as experimental and will break between release
 versions for the package that it is being imported from.
 
+### Testing
+
+#### Strategy
+
+In general we aim to test components from a user-focused perspective. This means
+avoiding testing implementation details, and instead focusing on writing tests
+that closely resemble how the components are used. The various `testing-library`
+packages are used to encourage this mindset when writing and composing test
+suites.
+
+#### Organization
+
+Every component should have tests covering a series of Categories
+
+- General component functionality/API
+- Accessibility
+- End to end tests
+- Server side rendering
+
+Each of these are separated into individual files. In some cases the syntax may
+be slightly different and separate files make this easier to understand.
+Additionally separate file types can be more easily globbed to only run certain
+tests in certain environments (local, CI, Pre-release checks, etc).
+
+| File name                    | Category                        |
+| ---------------------------- | ------------------------------- |
+| `ComponentName-test.js`      | General component functionality |
+| `ComponentName-test.a11y.js` | Accessibility testing           |
+| `ComponentName-test.e2e.js`  | End to end tests                |
+| `ComponentName-test.ssr.js`  | Server side rendering           |
+
+There are corresponding commands to run all categories, individual categories,
+or a combination. Depending on your shell, modifiers can be used to run two
+commands one after another. Refer to the documentation of your shell.
+
+| Command                           | Corresponding test category                                          |
+| --------------------------------- | -------------------------------------------------------------------- |
+| `yarn test`                       | All categories                                                       |
+| `yarn test:unit`                  | Only component unit tests                                            |
```

Comment:
Something like this? Was thinking we could add in during the beginnings of the refactor

```json
scripts: {
  "yarn test": "yarn test:unit && yarn test:a11y && ... maybe we look into running these in parallel rather than sequentially or something ",
  "yarn test:unit": "jest --testPathPattern='.*\\-test\\.js(x)?'",
  "yarn test:a11y": "jest --testPathPattern='.*\\.a11y\\.js(x)?'",
  "yarn test:e2e": "jest --testPathPattern='.*\\.e2e\\.js(x)?'",
  "yarn test:ssr": "jest --testPathPattern='.*\\.ssr\\.js(x)?'",
}
```